SalesforceCommerceCloud/pwa-kit
# @W-17271709 Passwordless Login in Checkout

# Description



Support the passwordless login flow when clicking the "Secure Link" button from the Checkout page.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- send `/oauth2/passwordless/login` API call when clicking the "Secure Link" button on the Checkout page

# How to Test-Drive This PR

- Navigate to https://wasatch-mrt-passwordless-poc.mrt-storefront-staging.com/
- Add an item to the cart and click *Proceed to Checkout*
- Under *Contact Info* leave the email field blank and click the "Secure Link" button
  - Verify a `Please enter your email address.` error msg is displayed
- Enter an invalid email like `j.ha` and click "Secure Link" button
  - Verify user is warned that an invalid email was entered
- Enter your salesforce email and click the "Secure Link" button
  - Verify a *Check Your Email* confirmation pop-up modal is displayed
  - Click the *Resend link* button 
- Check your email and verify 2 "Your Magic Link to Login" emails were sent and click on the the magic link within one of the emails
- Verify user is logged in and the cart still contains the item you added before login

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
